crates/ra_ide/src/hover.rs

```diff
@@ -1,16 +1,26 @@
+use std::collections::{HashMap, HashSet};
+use std::iter::once;
+
 use hir::{
-    Adt, AsAssocItem, AssocItemContainer, Documentation, FieldSource, HasSource, HirDisplay,
-    Module, ModuleDef, ModuleSource, Semantics,
+    db::DefDatabase, Adt, AsAssocItem, AsName, AssocItemContainer, AttrDef, Crate, Documentation,
```

Comment:
```suggestion
    Adt, AsAssocItem, AsName, AssocItemContainer, AttrDef, Crate, Documentation,
```

ide should not reach into DB direclty